### DISCUSSION TITLE
Tsyringe and vitest (esbuild options ignored?)

### DISCUSSION THREAD

**Discussion by Satont** [vitest-dev/vitest#1905] [Q&A]
With vitest resolved classes haven't needed dependencies, while normal file just works fine.

```ts
@injectable()
class MyApiClient {
 test = 123
 }

@singleton()
@autoInjectable()
export class TwitchAnnouncer {
  constructor(
    private readonly twitch: MyApiClient,
   ) {}
  }
  ```
  
  Vite:
  ```ts
  import { container } from 'tsyringe';
import { test } from 'vitest';
import { TwitchAnnouncer } from './src/announcers/twitchAnnouncer.js';
test('Test twitch announcer online', async () => {
  console.log(container.resolve(TwitchAnnouncer));
  // TwitchAnnouncer { twitch: undefined }
})
```

Ts-node:
```ts
import 'reflect-metadata';
import { container } from 'tsyringe';
import { TwitchAnnouncer } from './src/announcers/twitchAnnouncer.js';

console.log(container.resolve(TwitchAnnouncer));
// TwitchAnnouncer { twitch: MyApiClient {} }
```

https://stackblitz.com/edit/vitest-dev-vitest-ptiqdk

if you run `npm run test:ts-node`, then ts node will run example file when container working as expecting.

 System:
    OS: Linux 5.15 Ubuntu 20.04.4 LTS (Focal Fossa)
    CPU: (16) x64 AMD Ryzen 7 2700 Eight-Core Processor
    Memory: 7.69 GB / 10.70 GB
    Container: Yes
    Shell: 5.8 - /opt/distrod/alias/usr/bin/zsh
  Binaries:
    Node: 18.4.0 - ~/.nvm/versions/node/v18.4.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v18.4.0/bin/yarn
    npm: 8.12.1 - ~/.nvm/versions/node/v18.4.0/bin/npm

Vitest: `0.22.0`

**Comment by sheremet-va** [comment]
Decorators with meta information don't work by default. See https://github.com/vitest-dev/vitest/issues/708

**Reply by Satont** [reply]
Thats actually doesnt help. 

1. it doesnt work with ESM due https://github.com/egoist/unplugin-swc/issues/30
2. even if ESM disabled i got undefined https://stackblitz.com/edit/vitest-dev-vitest-leee36

```ts
plugins: [swc.esbuild()],
```
